pkg/auth/oidc_connect.go

```diff
@@ -0,0 +1,331 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/allegro/bigcache"
+	"github.com/coreos/go-oidc"
+	"github.com/go-logr/logr"
+	"github.com/projectcontour/contour-authserver/pkg/config"
+	"github.com/projectcontour/contour-authserver/pkg/store"
+	"golang.org/x/oauth2"
+)
+
+const (
+	stateQueryParamName = "conauth"
+	oauthTokenName      = "contourtoken"
+)
+
+// OIDCConnect .
+type OIDCConnect struct {
+	Log        logr.Logger
+	OidcConfig *config.OIDCConfig
+	Cache      *bigcache.BigCache
+	HTTPClient *http.Client
+	provider   *oidc.Provider
+}
+
+// Implement interface.
+var _ Checker = &OIDCConnect{}
+
+//Check ...
```

Comment:
Please add a comment describing what the function does, and its inputs and outputs.